### PR TITLE
fix: back button in detail view should not navigate to create dialog again

### LINKED ISSUE #5069: [Bug]: Back button after creating a client app
As an administrator I want to get to the project detail in the console when ever I click back from the application detail page.

## Acceptance Criteria
- [ ] I land on the project detail when clicking back (<--) on the application detail
- [ ] I land on the project detail when clicking back (<--) on the application detail when i just created the application

### Environment

ZITADEL Cloud

### Describe the bug

Inconsistent behavior on the back button when creating a new client application. After creating a new client and go on the back button the create app dialog starts again. However, editing the client and going back lands on the Project overview.

### To reproduce

Given I just created a client application (any type) on a project
When I click the back arrow
Then I get back to the create application dialog

Given I have already created an client application
When I go to my project and 
When I click on the application and
When I click the back button
Then I land on the project overview

### Screenshots

Screenshot 2023-01-17 at 17 10 23
Screenshot 2023-01-17 at 17 10 32

### Expected behavior

 I think the behavior should be in both cases to land on the project overview.

### PR DESCRIPTION
**Pull request by doncicuto**
@mffap detected in #5069 that after creating a new client and go on the back button the create app dialog starts again.

This is a video showing the current situation:

https://github.com/zitadel/zitadel/assets/30386061/4a0f8b97-38b0-4f18-8452-1328a8922924

The same behavior should apper after creating a project, a user, a machine user and a project role. As a workaround once of those elements are created I suggest adding a query param: new=true. When the back directive is applied after clicking the back button if the new param exists, a new back navigation is performed so the create dialog is not shown. Maybe is not the most elegant solution with going back twice.

This is a video showing the proposed change:

https://github.com/zitadel/zitadel/assets/30386061/57bc2855-0725-4feb-b315-49d56dd1f22d

This PR should close #5069.

Acceptance Criteria
- [X] I land on the project detail when clicking back (<--) on the application detail
- [X] I land on the project detail when clicking back (<--) on the application detail when i just created the application

### Definition of Ready

- [X] I am happy with the code
- [X] Short description of the feature/issue is added in the pr description
- [X] PR is linked to the corresponding user story
- [X] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [X] No debug or dead code
- [X] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [X] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Comment by hifabienne** [comment]
@doncicuto Thanks for the pr.
@peintnermax can you have a look at it?

**Review by peintnermax — CHANGES_REQUESTED** [review]
nice work 👍 . I've just added a small change such that the subscription of the params is cancelled.

File: console/src/app/directives/back/back.directive.ts [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
 
-  constructor(private navigation: NavigationService, private elRef: ElementRef, private renderer2: Renderer2) {
+  constructor(
+    private navigation: NavigationService,
+    private elRef: ElementRef,
+    private renderer2: Renderer2,
+    private route: ActivatedRoute,
+  ) {
+    // Check if a new element was created using a create dialog
+    this.route.queryParams.subscribe((params) => {
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/app/directives/back/back.directive.ts]: Change applied!

**Comment by doncicuto** [comment]
Thanks a lot @peintnermax for your review, changes have been applied

**Review by peintnermax — APPROVED** [review]

**Pull request merged**